workspace_tools/toolchains/arm.py

```diff
@@ -133,6 +133,7 @@ def link(self, output, objects, libraries, lib_dirs, mem_map):
 
         if mem_map:
             args.extend(["--scatter", mem_map])
+            args.extend(['--predefine="-D%s"' % s for s in self.get_symbols()])
```

Comment:
This is to pass the precompiler symbol to the scatter file so you can use the #ifdef SECONDARY_LOADER there.
I think it would be a bit overkill to create a copy for the whole LPC1768 just for the 3 lines difference in sct/s files, that's why my aim was to make it work with a compiler flag (that defaults to normal MBED if not set).